### ISSUE TITLE
output from HRnet+OCR

### ISSUE THREAD

**Issue by MandySuperTT** [HRNet/HRNet-Semantic-Segmentation#125]
Hi all,

I noticed that in `HRnet+OCR` branch, the output from the model is a list which contains 2 elements. However, it's not clear how these two elements work or their jobs in the loss functions and the following processing. 

I was wondering if there's anyone could give me explaination how does output be used in the loss function.

The original code is :

# seg_hrnet_ocr
        out_aux_seg = []

        # ocr
        out_aux = self.aux_head(feats)
        # compute contrast feature
        feats = self.conv3x3_ocr(feats)

        context = self.ocr_gather_head(feats, out_aux)
        feats = self.ocr_distri_head(feats, context)

        out = self.cls_head(feats)

        out_aux_seg.append(out_aux)
        out_aux_seg.append(out)

        return out_aux_seg

 # utils

    outputs = self.model(inputs, *args, **kwargs)
    loss = self.loss(outputs, labels)

One more question, why the `pred` from model don't need to be upsampled to the original image size during `training`?

**Comment by jcreinhold** [comment]
First, thank you for providing the package and great work!

For what it's worth, I'd also appreciate how to interpret the two outputs of the HRNet-OCR model. Which should be interpreted as the output? If I follow the code, it appears that the `out` variable should be interpreted as the output, but I'd like to verify that is the case. It does seem that `out` would make more sense as the final output.

@MandySuperTT Regarding your last question, the two outputs of `seg_hrnet_ocr` are upsampled inside the loss function defined [here](https://github.com/HRNet/HRNet-Semantic-Segmentation/blob/HRNet-OCR/lib/core/criterion.py).

**Comment by PkuRainBow** [comment]
@jcreinhold Thanks for your nice explanation for the issue of @MandySuperTT.

**Comment by daodao1991** [comment]
@MandySuperTT  
Hi, I think  yu should understood like this.
The first one of  output list belongs to the backbone network, which is used to calculate auxiliary loss. The second one belongs to OCR partition's output, and also is the last output of the whole network. In addition, the weights of this two part on total loss is different conrresponding config.LOSS.BALANCE_WEIGHTS in the config file.

**Comment by PkuRainBow** [comment]
@daodao1991 Thanks for your help. 
We recommend @MandySuperTT to check more details in our paper (ECCV-2020 version).
